content/en/blog/2022-03-21-putting-the-git-into-gitops/index.md

```diff
@@ -0,0 +1,189 @@
+enables new use-cases, support for more GPG verification or new SSH key
+formats adds additional security and when new features are rolled out in
+Git providers, we need to support them in Flux too.
+
+Integrating these changes into Flux unfortunately isn't as easy as it
+sounds. One part of the dependency chain for
+[git2go](https://github.com/libgit2/git2go) goes:
+
+```mermaid
+flowchart LR
+libgit2 --> libssh2 --> OpenSSL
+```
+
+So that's [libgit2](https://libgit2.org/),
+[libssh2](https://libssh2.org) (to enable the SSH transport) and
+[OpenSSL](https://www.openssl.org/). As Linux vendors often take a very
+conservative approach to bringing new software releases to stable
+releases, we were unfortunately pushed to
+[building these dependencies ourselves](https://github.com/fluxcd/golang-with-libgit2#rationale).
+In addition to that, these libraries have quite a few configuration
+options that can only be set at build time and unfortunately the
+`openssl`/`libssh2` packages of different Linux distributions act in
+[slightly different
+ways](https://github.com/fluxcd/golang-with-libgit2/blob/main/hack/Makefile#L63-L69).
+This created a yet different problem: the versions we shipped on the
+containers could behave in different ways when we were developing on our
+Mac/Linux machines. This forced us to cross-compile statically built
+libraries that we can simply download at both development time or
+statically linking them into the final binary we create when releasing
+our controllers.
+
+We decided to build the libraries for the AMD64, ARM64 and ARMv7
+architectures and link them statically, which was a prerequisite for us
+to [enable
+fuzzing](/blog/2022/02/security-more-confidence-through-fuzzing/)
+for all Flux controllers. Getting this all up and running for every
+upstream release and making sure it's all covered nicely with tests is a
+challenge and an area of work we invested quite a lot of time in.
+
+## What's next in Git things?
+
+`libgit2` does not expose concepts to allow users to set timeouts for
+network operations, meaning that most git operations could hang
+indefinitely in specific circumstances. This would result in specific
+`GitRepository` objects getting stuck and stop updating until the
+controllers get restarted - Users reported this for both the
+image-automation and source controllers in the past 6 months.
+
+We had a few challenges getting traction making changes upstream to fix
+similar issues, so to avoid delaying a fix or forking the dependency, we
+decided to add experimental support for Go managed transports, which
+we can enforce that network operations won't take more than a given
+amount of time to complete, but without requiring any changes upstream.
+
+This is [part of Flux 0.28](https://github.com/fluxcd/flux2/releases/tag/v0.28.0)
+and can be enabled by adding an environment `EXPERIMENTAL_GIT_TRANSPORT=true`
+in both `source` and `image-automation` controllers.
```

Comment:
Great advice from @stefanprodan for us to add an example on how folks can enable automatically as part of their clusters bootstrapping. Add into the `kustomization.yaml`:

```yaml
patches:
- patch: |
    - op: add
      path: /spec/template/spec/containers/0/env/0
      value: 
        name: EXPERIMENTAL_GIT_TRANSPORT
        value: "true"
  target:
    kind: Deployment
    name: "(source-controller|image-automation-controller)"

```